runtime/compiler/aarch64/codegen/J9TreeEvaluator.cpp

```diff
@@ -5950,6 +5951,263 @@ static TR::Register *inlineCRC32CUpdateBytes(TR::Node *node, TR::CodeGenerator *
+    *
+    *    If the number of characters in the string is not a multiple of 16 (or 8 in the case of decompressed String),
+    *    then the remainder of the hash is calculated serially.
+    *
+    *	   addx dataAddrReg, addressReg, #headersize
+    *    movz resultReg, #0
+    *    cmpw lengthReg, #(8 or 16)
+    *    b.lt residual
+    *    sub lengthReg, lengthReg, #(8 or 16)
+    *    ldr vMultiplierReg, multiplier
+    *    ldr vMultiplier4Reg, multiplier4
+    *    mul vMultiplier8Reg.4s, vMultiplier4Reg.4s, vMultiplier4Reg.4s ; 31^8
+    *    movi vtmp1Reg.16b, #0
+    *    movi vtmp2Reg.16b, #0
+    * LOOP:
+    *    subs lengthReg, lengthReg, #(8 or 16)
+    *    ldr vtmp3Reg, [dataAddrReg], #16
+    *
+    * if decompressed
+    *    uxtl vtmp4Reg.4s, vtmp3Reg.4h
+    *    uxtl2 vtmp5Reg.4s, vtmp3Reg.8h
+    *    ; (1) Vector0[k] = str[k] -> multiply 31^8 -> add str[k+8]
+    *    mla vtmp4Reg.4s, vtmp1Reg.4s, vMultiplier8Reg.4s
+    *    mov vtmp1Reg.16b, vtmp4Reg.16b
+    *    ; (2) Vector1[k] = str[k+4] -> multiply 31^8 -> add str[k+12]
+    *    mla vtmp5Reg.4s, vtmp2Reg.4s, vMultiplier8Reg.4s
+    *    mov vtmp2Reg.16b, vtmp5Reg.16b
+    * else ; compressed
+    *    uxtl vtmp4Reg.8h, vtmp3Reg.8b
+    *    uxtl2 vtmp5Reg.8h, vtmp3Reg.16b
+    *    uxtl vtmp6Reg.4s, vtmp4Reg.4h
+    *    uxtl2 vtmp3Reg.4s, vtmp4Reg.8h
+    *    ; (1) Vector0[k] = str[k] -> multiply 31^8 -> add str[k+8]
+    *    mla vtmp6Reg.4s, vtmp1Reg.4s, vMultiplier8Reg.4s
+    *    uxtl vtmp1Reg.4s, vtmp5Reg.4h
+    *    ; (2) Vector1[k] = str[k+4] -> multiply 31^8 -> add str[k+12]
+    *    mla vtmp3Reg.4s, vtmp2Reg.4s, vMultiplier8Reg.4s
+    *    uxtl2 vtmp2Reg.4s, vtmp5Reg.8h
+    *    ; (1) Vector0[k] = str[k] -> multiply 31^8 -> add str[k+8]
+    *    mla vtmp1Reg.4s, vtmp6Reg.4s, vMultiplier8Reg.4s
+    *    ; (2) Vector1[k] = str[k+4] -> multiply 31^8 -> add str[k+12]
+    *    mla vtmp2Reg.4s, vtmp3Reg.4s, vMultiplier8Reg.4s
+    * fi
+    *    b.ge LOOP
+    *    ; (3) Vector[k] = Vector0[k] -> multiply 31^4 -> add Vector1[k]
+    *    mla vtmp2Reg.4s, vtmp1Reg.4s, vMultiplier4Reg.4s
+    *    ; (3) -> multiply 31^(3-k)
+    *    mul vtmp1Reg.4s, vtmp2Reg.4s, vMultiplierReg.4s
+    *    addv vtmp1Reg, vtmp1Reg.4s
+    *    mov resultReg, vtmp1Reg.s[0]
+    *    addws lengthReg, lengthReg, #(8 or 16)
+    *    b.eq done
+    * residual:
+    *    subws lengthReg, lengthReg, #1
+    * if decompressed
+    *    ldrh dataReg, [dataAddrReg], #2
+    * else ; compressed
+    *    ldrb dataReg, [dataAddrReg], #1
+    * fi
+    *    addw dataReg, dataReg, resultReg, lsl #5
+    *    subw resultReg, dataReg, resultReg
+    *    b.gt residual
+    * done:
+    *    ret
+    *    .align 3
+    * multiplier:
+    *    .dword 0x3c10000745f   ; 31^2, 31^3
+    *    .dword 0x10000001f     ;    1, 31
+    * multiplier4:
+    *    .dword 0xe1781000e1781 ; 31^4
+    *    .dword 0xe1781000e1781
```

Comment:
Maybe full digit representation such as `0x000003c10000745f` is better for these lines.

Reply:
Changed them to full hex digit representation.